tools/lsp/util.rs

```diff
@@ -44,15 +47,16 @@ pub fn map_range(sf: &SourceFile, range: TextRange) -> lsp_types::Range {
     lsp_types::Range::new(map_position(sf, range.start()), map_position(sf, range.end()))
 }
 
-pub fn map_to_offset(sf: &SourceFile, position: lsp_types::Position) -> usize {
-    sf.offset(
+pub fn map_text_size(sf: &SourceFile, position: lsp_types::Position) -> TextSize {
+    (sf.offset(
         usize::try_from(position.line).unwrap() + 1,
         usize::try_from(position.character).unwrap() + 1,
-    )
+    ) as u32)
+        .into()
 }
 
-pub fn map_to_offsets(sf: &SourceFile, range: lsp_types::Range) -> (usize, usize) {
-    (map_to_offset(sf, range.start), map_to_offset(sf, range.end))
+pub fn map_text_range(sf: &SourceFile, range: lsp_types::Range) -> TextRange {
```

Comment:
This is also strange name as it maps a `lsp_types::Range` to a `rowan::TextRange`, while the `map_range` funciton do the other one but that a similar name.

Maybe all these function should be called something like `from_lsp_range` / `to_lsp_range` / `from_lsp_position` / `to_lsp_position`

Or there could even be a `trait MapLspType { type From; fn from_lsp(_:From, sf:&SourceFile)->Self; fn into_lsp(&self,sf:&SourceFile)->From;}` but maybe that's overkill.